src/shared/dom.js

```diff
@@ -22,6 +22,20 @@ export function teardownEach ( iterations, detach, start ) {
 	}
 }
 
+export function spreadAttributes ( node, obj ) {
+	if (obj !== null && typeof obj === 'object') {
+		for ( var property in obj ) {
```

Comment:
Oh and on browser compatibility, the only browser with significant market share that doesn't support ES5 is IE8 at 3% marketshare. But I honestly think that 8 years later we can ship ES5 features to browsers. + there's plenty of polyfills for ES5 at this point.